src/upchunk.ts

```diff
@@ -128,7 +128,7 @@ export class UpChunk {
   private dispatch(eventName: EventName, detail?: any) {
     const event = new CustomEvent(eventName, { detail });
 
-    this.eventTarget.dispatchEvent(event);
+    this.eventTarget.dispatchEvent(event as any);
```

Comment:
Could this be `CustomEvent` instead of `any`?

Reply:
Thank you so much for the quick review!

it's not working for me -- here are the error messages I get:

```bash
barakgila@Baraks-MacBook-Pro upchunk (bg-update-event-target-shim) $ yarn build
yarn run v1.22.17
$ yarn clean && yarn lint && yarn test && webpack --config webpack.prod.js && yarn pack
$ rm -rf ./dist
$ tslint --project .
no-unused-variable is deprecated. Since TypeScript 2.9. Please use the built-in compiler checks instead.
$ jest
ts-jest[versions] (WARN) Version 4.6.3 of typescript installed has not been tested with ts-jest. If you're experiencing issues, consider using a supported version (>=3.8.0 <4.0.0-0). Please do not report issues in ts-jest if you are using unsupported versions.
ts-jest[versions] (WARN) Version 4.6.3 of typescript installed has not been tested with ts-jest. If you're experiencing issues, consider using a supported version (>=3.8.0 <4.0.0-0). Please do not report issues in ts-jest if you are using unsupported versions.
 FAIL  src/upchunk.spec.js
  ● Test suite failed to run

    src/upchunk.ts:131:36 - error TS2769: No overload matches this call.
      Overload 1 of 2, '(event: never): boolean', gave the following error.
        Argument of type 'CustomEvent<any>' is not assignable to parameter of type 'never'.
      Overload 2 of 2, '(event: Event<string>): boolean', gave the following error.
        Argument of type 'CustomEvent<any>' is not assignable to parameter of type 'Event<string>'.
          Types of property 'target' are incompatible.
            Type 'EventTarget | null' is not assignable to type 'EventTarget<Record<string, Event<string>>, "standard"> | null'.
              Type 'EventTarget' is not assignable to type 'EventTarget<Record<string, Event<string>>, "standard">'.
                Types of property 'addEventListener' are incompatible.
                  Type '(type: string, callback: EventListenerOrEventListenerObject | null, options?: boolean | AddEventListenerOptions | undefined) => void' is not assignable to type '{ <T extends string>(type: T, callback?: EventListener<EventTarget<Record<string, Event<string>>, "standard">, Event<string>> | null | undefined, options?: AddOptions | undefined): void; (type: string, callback?: EventListener<...> | ... 1 more ... | undefined, options?: AddOptions | undefined): void; <T extends str...'.
                    Types of parameters 'options' and 'options' are incompatible.
                      Type 'AddOptions | undefined' is not assignable to type 'boolean | AddEventListenerOptions | undefined'.
                        Type 'AddOptions' is not assignable to type 'boolean | AddEventListenerOptions | undefined'.
                          Type 'AddOptions' is not assignable to type 'AddEventListenerOptions'.
                            Types of property 'signal' are incompatible.
                              Type 'AbortSignal | null | undefined' is not assignable to type 'AbortSignal | undefined'.
                                Type 'null' is not assignable to type 'AbortSignal | undefined'.

    131     this.eventTarget.dispatchEvent(event as CustomEvent);
                                           ~~~~~~~~~~~~~~~~~~~~

 FAIL  src/upchunk.spec.ts
  ● Test suite failed to run

    src/upchunk.ts:131:36 - error TS2769: No overload matches this call.
      Overload 1 of 2, '(event: never): boolean', gave the following error.
        Argument of type 'CustomEvent<any>' is not assignable to parameter of type 'never'.
      Overload 2 of 2, '(event: Event<string>): boolean', gave the following error.
        Argument of type 'CustomEvent<any>' is not assignable to parameter of type 'Event<string>'.
          Types of property 'target' are incompatible.
            Type 'EventTarget | null' is not assignable to type 'EventTarget<Record<string, Event<string>>, "standard"> | null'.
              Type 'EventTarget' is not assignable to type 'EventTarget<Record<string, Event<string>>, "standard">'.
                Types of property 'addEventListener' are incompatible.
                  Type '(type: string, callback: EventListenerOrEventListenerObject | null, options?: boolean | AddEventListenerOptions | undefined) => void' is not assignable to type '{ <T extends string>(type: T, callback?: EventListener<EventTarget<Record<string, Event<string>>, "standard">, Event<string>> | null | undefined, options?: AddOptions | undefined): void; (type: string, callback?: EventListener<...> | ... 1 more ... | undefined, options?: AddOptions | undefined): void; <T extends str...'.
                    Types of parameters 'options' and 'options' are incompatible.
                      Type 'AddOptions | undefined' is not assignable to type 'boolean | AddEventListenerOptions | undefined'.
                        Type 'AddOptions' is not assignable to type 'boolean | AddEventListenerOptions | undefined'.
                          Type 'AddOptions' is not assignable to type 'AddEventListenerOptions'.
                            Types of property 'signal' are incompatible.
                              Type 'AbortSignal | null | undefined' is not assignable to type 'AbortSignal | undefined'.
                                Type 'null' is not assignable to type 'AbortSignal | undefined'.

    131     this.eventTarget.dispatchEvent(event as CustomEvent);
                                           ~~~~~~~~~~~~~~~~~~~~

Test Suites: 2 failed, 2 total
Tests:       0 total
Snapshots:   0 total
Time:        1.61 s, estimated 2 s
Ran all test suites.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
barakgila@Baraks-MacBook-Pro upchunk (bg-update-event-target-shim) $ yarn build
yarn run v1.22.17
$ yarn clean && yarn lint && yarn test && webpack --config webpack.prod.js && yarn pack
$ rm -rf ./dist
$ tslint --project .
no-unused-variable is deprecated. Since TypeScript 2.9. Please use the built-in compiler checks instead.
$ jest
ts-jest[versions] (WARN) Version 4.6.3 of typescript installed has not been tested with ts-jest. If you're experiencing issues, consider using a supported version (>=3.8.0 <4.0.0-0). Please do not report issues in ts-jest if you are using unsupported versions.
ts-jest[versions] (WARN) Version 4.6.3 of typescript installed has not been tested with ts-jest. If you're experiencing issues, consider using a supported version (>=3.8.0 <4.0.0-0). Please do not report issues in ts-jest if you are using unsupported versions.
 FAIL  src/upchunk.spec.js
  ● Test suite failed to run

    src/upchunk.ts:131:36 - error TS2769: No overload matches this call.
      Overload 1 of 2, '(event: never): boolean', gave the following error.
        Argument of type 'Event' is not assignable to parameter of type 'never'.
      Overload 2 of 2, '(event: Event<string>): boolean', gave the following error.
        Argument of type 'Event' is not assignable to parameter of type 'Event<string>'.
          Types of property 'target' are incompatible.
            Type 'EventTarget | null' is not assignable to type 'EventTarget<Record<string, Event<string>>, "standard"> | null'.
              Type 'EventTarget' is not assignable to type 'EventTarget<Record<string, Event<string>>, "standard">'.
                Types of property 'addEventListener' are incompatible.
                  Type '(type: string, callback: EventListenerOrEventListenerObject | null, options?: boolean | AddEventListenerOptions | undefined) => void' is not assignable to type '{ <T extends string>(type: T, callback?: EventListener<EventTarget<Record<string, Event<string>>, "standard">, Event<string>> | null | undefined, options?: AddOptions | undefined): void; (type: string, callback?: EventListener<...> | ... 1 more ... | undefined, options?: AddOptions | undefined): void; <T extends str...'.
                    Types of parameters 'options' and 'options' are incompatible.
                      Type 'AddOptions | undefined' is not assignable to type 'boolean | AddEventListenerOptions | undefined'.
                        Type 'AddOptions' is not assignable to type 'boolean | AddEventListenerOptions | undefined'.
                          Type 'AddOptions' is not assignable to type 'AddEventListenerOptions'.
                            Types of property 'signal' are incompatible.
                              Type 'AbortSignal | null | undefined' is not assignable to type 'AbortSignal | undefined'.
                                Type 'null' is not assignable to type 'AbortSignal | undefined'.

    131     this.eventTarget.dispatchEvent(event as Event);
                                           ~~~~~~~~~~~~~~

 FAIL  src/upchunk.spec.ts
  ● Test suite failed to run

    src/upchunk.ts:131:36 - error TS2769: No overload matches this call.
      Overload 1 of 2, '(event: never): boolean', gave the following error.
        Argument of type 'Event' is not assignable to parameter of type 'never'.
      Overload 2 of 2, '(event: Event<string>): boolean', gave the following error.
        Argument of type 'Event' is not assignable to parameter of type 'Event<string>'.
          Types of property 'target' are incompatible.
            Type 'EventTarget | null' is not assignable to type 'EventTarget<Record<string, Event<string>>, "standard"> | null'.
              Type 'EventTarget' is not assignable to type 'EventTarget<Record<string, Event<string>>, "standard">'.
                Types of property 'addEventListener' are incompatible.
                  Type '(type: string, callback: EventListenerOrEventListenerObject | null, options?: boolean | AddEventListenerOptions | undefined) => void' is not assignable to type '{ <T extends string>(type: T, callback?: EventListener<EventTarget<Record<string, Event<string>>, "standard">, Event<string>> | null | undefined, options?: AddOptions | undefined): void; (type: string, callback?: EventListener<...> | ... 1 more ... | undefined, options?: AddOptions | undefined): void; <T extends str...'.
                    Types of parameters 'options' and 'options' are incompatible.
                      Type 'AddOptions | undefined' is not assignable to type 'boolean | AddEventListenerOptions | undefined'.
                        Type 'AddOptions' is not assignable to type 'boolean | AddEventListenerOptions | undefined'.
                          Type 'AddOptions' is not assignable to type 'AddEventListenerOptions'.
                            Types of property 'signal' are incompatible.
                              Type 'AbortSignal | null | undefined' is not assignable to type 'AbortSignal | undefined'.
                                Type 'null' is not assignable to type 'AbortSignal | undefined'.

    131     this.eventTarget.dispatchEvent(event as Event);
                                           ~~~~~~~~~~~~~~

Test Suites: 2 failed, 2 total
Tests:       0 total
Snapshots:   0 total
Time:        1.583 s
Ran all test suites.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Reply:
Hmm ok well I don't think that should block this from getting merged, so I'll merge and revisit after. Thanks again for the contribution!

Reply:
No problem, thank you! I'll stay tuned for your merge/version release processes